server/polar/refund/endpoints.py

```diff
@@ -0,0 +1,113 @@
+from .schemas import RefundCreate, RefundID
+from .service import RefundAmountTooHigh, RefundedAlready
+from .service import refund as refund_service
+from .sorting import RefundListSorting
+
+log = structlog.get_logger()
+
+router = APIRouter(prefix="/refunds", tags=["refunds", APITag.documented])
+
+OrderNotFound = {
+    "description": "Order not found.",
+    "model": ResourceNotFound.schema(),
+}
+
+
+@router.get("/", summary="List Refunds", response_model=ListResource[RefundSchema])
+async def list(
+    pagination: PaginationParamsQuery,
+    sorting: RefundListSorting,
+    auth_subject: auth.RefundsRead,
+    id: MultipleQueryFilter[RefundID] | None = Query(
+        None, title="RefundID Filter", description="Filter by refund ID."
+    ),
+    organization_id: MultipleQueryFilter[OrganizationID] | None = Query(
+        None, title="OrganizationID Filter", description="Filter by organization ID."
+    ),
+    order_id: MultipleQueryFilter[OrderID] | None = Query(
+        None, title="OrderID Filter", description="Filter by order ID."
+    ),
+    subscription_id: MultipleQueryFilter[SubscriptionID] | None = Query(
+        None, title="SubscriptionID Filter", description="Filter by subscription ID."
+    ),
+    customer_id: MultipleQueryFilter[CustomerID] | None = Query(
+        None, title="CustomerID Filter", description="Filter by customer ID."
+    ),
+    succeeded: bool = Query(
+        True,
+        title="RefundStatus Filter",
+        description="Filter by `succeeded` or not.",
+    ),
+    session: AsyncSession = Depends(get_db_session),
+) -> ListResource[RefundSchema]:
+    """List products."""
+    results, count = await refund_service.get_list(
+        session,
+        auth_subject,
+        id=id,
+        organization_id=organization_id,
+        order_id=order_id,
+        subscription_id=subscription_id,
+        customer_id=customer_id,
+        succeeded=succeeded,
+        pagination=pagination,
+        sorting=sorting,
+    )
+
+    return ListResource.from_paginated_results(
+        [RefundSchema.model_validate(result) for result in results],
+        count,
+        pagination,
+    )
+
+
+@router.post(
+    "/",
+    summary="Create Refund",
+    response_model=RefundSchema,
+    responses={
+        201: {"description": "Refund created."},
+        400: {
+            "description": "Refund amount exceeds remaining order balance.",
+            "model": RefundAmountTooHigh.schema(),
+        },
+        403: {
+            "description": "Order is already fully refunded.",
+            "model": RefundedAlready.schema(),
+        },
+        404: OrderNotFound,
+    },
+)
+async def create(
+    refund_create: RefundCreate,
+    auth_subject: auth.RefundsWrite,
+    session: AsyncSession = Depends(get_db_session),
+) -> Refund:
+    """Create a refund."""
+    order = await order_service.user_get_by_id(
+        session,
+        auth_subject,
+        refund_create.order_id,
+    )
+    if not order:
```

Comment:
Kind of a nitpick, but I would move this to the `create` method and raise a `PolarRequestValidationError` (422) instead of a 404.

Reply:
Fixing + getting rid of 404/OrderNotFound